packages/babel-plugin-transform-class-properties/src/index.js

```diff
@@ -85,9 +85,9 @@ export default function({ types: t }) {
           const propNode = prop.node;
           if (propNode.decorators && propNode.decorators.length > 0) continue;
 
-          // In non-spec mode, all properties without values are ignored.
+          // In loose mode, all properties without values are ignored.
           // In spec mode, *static* properties without values are still defined (see below).
```

Comment:
IIUC, in spec mode, _all_ properties should be defined. This may also be a new change on the stage-3 version.

* https://tc39.github.io/proposal-class-fields/#runtime-semantics-class-field-definition-evaluation the Record is still created even if the Initializer is missing
* https://tc39.github.io/proposal-class-fields/#initialize-public-instance-fields InitializeInstanceFields does not care whether the record has an initializer
* https://tc39.github.io/proposal-class-fields/#sec-define-field DefineField will use an initValue of undefined if there is no initializer